src/interface/blas3/backend/amd_gpu.hpp

```diff
@@ -54,58 +54,60 @@ typename sb_handle_t::event_t _gemm(
-          _t_a, _t_b, s_a, s_b, static_cast<int>(gemm_memory_t::local),
-          static_cast<int>(gemm_algorithm_t::tall_skinny),
-          static_cast<int>(gemm_vectorization_t::none), is_beta_zero, 2,
-          static_cast<int>(gemm_batch_type_t::strided)>::
-          template _select_gemm(sb_handle, _M, _N, _K, _alpha, _a, _lda,
-                                _stridea, _b, _ldb, _strideb, _beta, _c, _ldc,
-                                _stridec, batch_size);
-    } else if (_M > 64 && _N <= 32) {
-      return blas::Gemm_Launcher<
-          256, true, true, true, ClSize, Tile<4, 1, tileWgSize, tileWgSize>,
-          _t_a, _t_b, s_a, s_b, static_cast<int>(gemm_memory_t::local),
-          static_cast<int>(gemm_algorithm_t::tall_skinny),
-          static_cast<int>(gemm_vectorization_t::none), is_beta_zero, 2,
-          static_cast<int>(gemm_batch_type_t::strided)>::
-          template _select_gemm(sb_handle, _M, _N, _K, _alpha, _a, _lda,
-                                _stridea, _b, _ldb, _strideb, _beta, _c, _ldc,
-                                _stridec, batch_size);
-    } else if (_M <= 16 || _N <= 16) {
-      return blas::Gemm_Launcher<
-          256, true, true, true, ClSize, Tile<1, 1, tileWgSize, tileWgSize>,
-          _t_a, _t_b, s_a, s_b, static_cast<int>(gemm_memory_t::local),
-          static_cast<int>(gemm_algorithm_t::tall_skinny),
-          static_cast<int>(gemm_vectorization_t::none), is_beta_zero, 2,
-          static_cast<int>(gemm_batch_type_t::strided)>::
-          template _select_gemm(sb_handle, _M, _N, _K, _alpha, _a, _lda,
-                                _stridea, _b, _ldb, _strideb, _beta, _c, _ldc,
-                                _stridec, batch_size);
-    } else if (_M <= 32 || _N <= 32) {
-      return blas::Gemm_Launcher<
-          256, true, true, true, ClSize, Tile<2, 2, tileWgSize, tileWgSize>,
-          _t_a, _t_b, s_a, s_b, static_cast<int>(gemm_memory_t::local),
-          static_cast<int>(gemm_algorithm_t::tall_skinny),
-          static_cast<int>(gemm_vectorization_t::none), is_beta_zero, 2,
-          static_cast<int>(gemm_batch_type_t::strided)>::
-          template _select_gemm(sb_handle, _M, _N, _K, _alpha, _a, _lda,
-                                _stridea, _b, _ldb, _strideb, _beta, _c, _ldc,
-                                _stridec, batch_size);
-    } else {
-      return blas::Gemm_Launcher<
-          256, true, true, true, ClSize, Tile<4, 4, tileWgSize, tileWgSize>,
-          _t_a, _t_b, s_a, s_b, static_cast<int>(gemm_memory_t::local),
-          static_cast<int>(gemm_algorithm_t::tall_skinny),
-          static_cast<int>(gemm_vectorization_t::none), is_beta_zero, 2,
-          static_cast<int>(gemm_batch_type_t::strided)>::
-          template _select_gemm(sb_handle, _M, _N, _K, _alpha, _a, _lda,
-                                _stridea, _b, _ldb, _strideb, _beta, _c, _ldc,
-                                _stridec, batch_size);
+  if (!s_a && !s_b) {
```

Comment:
This if condition should be placed **after** the `if (batch_size == 1 && ...)` otherwise it changes the logic behind the if/else conditions (line 58 - 112 - etc..).